src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5408,12 +5431,22 @@ address MacroAssembler::arrays_equals(Register a1, Register a2, Register tmp3,
     mov(result, a2);
     b(DONE);
     bind(SHORT);
-    cmp(cnt2, cnt1);
-    br(NE, DONE);
+    if (is_8aligned) {
+      cmp(cnt2, cnt1);
+      br(NE, DONE);
+    }
     cbz(cnt1, SAME);
```

Comment:
Was this an optimization? The `cbz` would never be taken if `!is_8aligned `  as it implies `cnt1 > 0`

Reply:
Yes. The idea is that if the main loop also compares the lenghts, then we don't need to explicitely compare them before the main loop and save a bunch of instructions. I removed the special-handling there for now, for simplicity.